vsketch/sketch_class.py

```diff
@@ -58,6 +59,15 @@ def ensure_finalized(self) -> None:
 
         self._finalized = True
 
+    def ensure_post_finalized(self, path: pathlib.Path) -> None:
+        # only do anything if sketch is finalized but not post_finalized
+        if not self._finalized or self._post_finalized:
+            return
+
+        self.post_finalize(self._vsk, path)
+
+        self._post_finalized = True
+
```

Comment:
Now that I see it, I have a doubt with the implementation of this function. Unlike `finalize` (which should be used only once for a given sketch), post_finalize must be called _every single time_ an output file is saved. In particular, if you click multiple times on the like button without changing the sketch call, `finalize` should be called once, but post_finalize should be called as many times as Like is clicked.

I think it would be more correct to remove that, and simply call `post_finalize` in the two places were you currently call `ensure_post_finalize`.

Reply:
Agreed.  This has been changed in the most recent commit. :)